staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go

```diff
@@ -137,15 +181,75 @@ func (m *PathRecorderMux) UnlistedHandle(path string, handler http.Handler) {
 // UnlistedHandleFunc registers the handler function for the given pattern, but doesn't list it.
 // If a handler already exists for pattern, Handle panics.
 func (m *PathRecorderMux) UnlistedHandleFunc(path string, handler func(http.ResponseWriter, *http.Request)) {
+	m.UnlistedHandle(path, http.HandlerFunc(handler))
+}
+
+// HandlePrefix is like Handle, but matches for anything under the path.  Like a standard golang trailing slash.
+func (m *PathRecorderMux) HandlePrefix(path string, handler http.Handler) {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 	m.trackCallers(path)
 
-	m.pathToHandler[path] = http.HandlerFunc(handler)
+	m.exposedPaths = append(m.exposedPaths, path)
+	m.prefixToHandler[path] = handler
+	m.refreshMuxLocked()
+}
+
+// UnlistedHandlePrefix is like UnlistedHandle, but matches for anything under the path.  Like a standard golang trailing slash.
+func (m *PathRecorderMux) UnlistedHandlePrefix(path string, handler http.Handler) {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+	m.trackCallers(path)
+
+	m.prefixToHandler[path] = handler
 	m.refreshMuxLocked()
 }
 
 // ServeHTTP makes it an http.Handler
 func (m *PathRecorderMux) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	m.mux.Load().(*http.ServeMux).ServeHTTP(w, r)
+	m.mux.Load().(pathHandler).ServeHTTP(w, r)
+}
+
+// ServeHTTP makes it an http.Handler
+func (h pathHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
> any reason not to use a pointer as receiver here?

An object with three pointers?  I'll update, but I didn't think it really mattered.